kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/notification/WebSocketNotificationService.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.controller.websocket.notification;
+
+import java.util.List;
+
+import org.kie.server.controller.api.model.events.*;
+import org.kie.server.controller.api.model.notification.KieServerControllerNotification;
+import org.kie.server.controller.api.model.runtime.Container;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.controller.api.service.NotificationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WebSocketNotificationService implements NotificationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WebSocketNotificationService.class);
+
+    private static WebSocketNotificationService INSTANCE = new WebSocketNotificationService();
+
+    protected WebSocketNotificationService() {
+        LOGGER.info("WebSocket Notification Service started.");
+    }
+
+    public static WebSocketNotificationService getInstance() {
+        return INSTANCE;
+    }
+
+    private void notifySessions(final KieServerControllerEvent event) {
+        final KieServerControllerNotification notification = new KieServerControllerNotification(event);
+
+        LOGGER.debug("Sending notification to all Web Socket sessions");
+        WebSocketNotificationSessionManager.getInstance().getSessions().forEach(
+                session -> {
+                    LOGGER.debug("Sending notification to Web Socket session with id: {}",
+                                 session.getId());
+                    session.getAsyncRemote().sendObject(notification,
```

Comment:
error handling in the for each (or replace for each with regular for loop) would be good. As in case one of the sessions fails it will skip notifications to other too.

Reply:
that's probably not necessary as each send is done by a different thread using the asyncRemote. Handling is done individually with the _SendHandler_ to hold any exception.

Reply:
assuming sendObject on asyncRemote won't throw exception, but I see your point and it should be enough, thanks